Source/scriptfunc.lua

```diff
@@ -1309,3 +1309,33 @@ function delete_script(script_i)
 	-- We might have removed a script reference, so update usages
 	usedscripts, n_usedscripts = find_used_scripts()
 end
+
+function get_valid_commands()
+	-- Get the commands for this context.
+	local commands = {}
+
+	local internal = (internalscript or cutscenebarsinternalscript)
+
+	for command, _ in pairs(internal and knowninternalcommands or knowncommands) do
+		table.insert(commands, command)
+	end
+
+	return commands
+end
+
+function get_autocomplete_commands(command_part)
+	-- Return a sorted list of commands that start with command_part
+	local commands = {}
+	for _, command in pairs(get_valid_commands()) do
+		if command:sub(1, #command_part) == command_part then
```

Comment:
I think normally Ved uses `s:len()` instead of `#s` for strings, and it'd be more optimized to do this above the loop (so that'd be another improvement over the original code which also did it that way)